### PR TITLE
Map LevelGeneratorOptions

### PR DESCRIPTION
**Pull request by falseresync**
Credit: liach for corrections and extension

### CODE REVIEW

**Comment by falseresync** [comment]
Whoops

**Review by Runemoro — APPROVED** [review]

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 1]
```diff
@@ -0,0 +1,25 @@
+CLASS net/minecraft/class_4952 net/minecraft/world/level/LevelGeneratorOptions
+	FIELD field_23085 generatorType Lnet/minecraft/class_1942;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]:
```suggestion
	FIELD field_23085 type Lnet/minecraft/class_1942;
```

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	FIELD field_23086 dynamic Lcom/mojang/datafixers/Dynamic;
+	FIELD field_23087 generatorProducer Ljava/util/function/Function;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]:
```suggestion
	FIELD field_23087 chunkGeneratorFactory Ljava/util/function/Function;
```

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	METHOD <init> (Lnet/minecraft/class_1942;Lcom/mojang/datafixers/Dynamic;Ljava/util/function/Function;)V
+		ARG 1 generatorType
+		ARG 3 generatorProducer
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]:
```suggestion
		ARG 1 type
		ARG 2 dynamic
		ARG 3 chunkGeneratorFactory
```

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	METHOD method_25963 getGeneratorType ()Lnet/minecraft/class_1942;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]:
```suggestion
	METHOD method_25963 getType ()Lnet/minecraft/class_1942;
```

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	METHOD method_25966 getGenerator (Lnet/minecraft/class_1936;)Lnet/minecraft/class_2794;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]:
```suggestion
	METHOD method_25966 createChunkGenerator (Lnet/minecraft/class_1936;)Lnet/minecraft/class_2794;
```

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+		ARG 1 world
+	METHOD method_25968 createDefault (Lnet/minecraft/class_1942;Lcom/mojang/datafixers/Dynamic;)Lnet/minecraft/class_4952;
+		ARG 0 generatorType
+	METHOD method_25971 setDefaultStates (Lnet/minecraft/class_2888;Lcom/mojang/datafixers/DynamicLike;)V
+	METHOD method_25974 getBiomeSource (Lcom/mojang/datafixers/DynamicLike;)Ljava/util/function/LongFunction;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]: This one is wrong... Let me find a better name for it

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	METHOD method_25975 createCustom (Lcom/mojang/datafixers/DynamicLike;Lnet/minecraft/class_2798;Ljava/util/function/LongFunction;)Ljava/util/function/Function;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]: Not right either

File: mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	METHOD method_25976 retrieveFromRegistry (Lcom/mojang/datafixers/DynamicLike;Lnet/minecraft/class_2378;Ljava/lang/Object;)Ljava/lang/Object;
+		ARG 2 defaultObject
+	METHOD method_25977 getGenerator (Lcom/mojang/datafixers/DynamicLike;Ljava/util/function/LongFunction;)Ljava/util/function/Function;
```
Comment (liach) [inline mappings/net/minecraft/world/level/LevelGeneratorOptions.mapping]:
```suggestion
	METHOD method_25977 createBuffetChunkGeneratorFactory (Lcom/mojang/datafixers/DynamicLike;Ljava/util/function/LongFunction;)Ljava/util/function/Function;
```

**Review by jaskarth — APPROVED** [review]

**Comment by liach** [comment]
After a look in enigma, I find this pr... a bit inaccurate in its names

**Comment by falseresync** [comment]
Yes, I know it is, it was meant to be a draft (I don’t know how to make a draft PR if that’s possible). I want to map it, so this PR is a kind of a claim :P

**Comment by liach** [comment]
Opened a pr falseresync/yarn#1 with some of my mappings wanted as this gen options has associated other changes

**Comment by falseresync** [comment]
Thank you, liach

**Review by liach — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by jaskarth — APPROVED** [review]
looks good to me

**Review by Runemoro — APPROVED** [review]

**Pull request merged**